spec/book_spec.rb

```diff
@@ -0,0 +1,24 @@
+require_relative '../lib/book'
+require_relative '../lib/item'
+
+describe Book do
+  describe 'when creating a new book' do
+    publish_date = '2023-04-25'
+    publisher = 'Hans Rameau'
+    cover_state = 'bad'
+    book = Book.new(publish_date, publisher, cover_state)
+    book2 = Book.new('2023-01-01', publisher, 'good')
+
+    it 'should be an instance of BOOK' do
+      expect(book).to be_instance_of(Book)
+    end
+
+    it 'should return true if cover state equal to bad or can_be_archived? is true' do
+      expect(book.can_be_archived?).to be(true)
+    end
+
+    it 'should return false if cover state equal to good or can_be_archived? is true' do
+      expect(book2.can_be_archived?).to be(false)
+    end
+  end
+end
```

Comment:
Excellent implementation of your tests for the Book class 💯. However, please kindly make sure that you add a test for the `add_item` method of your Label class too as requested in the requirement below 👇🏽.

- All methods visible in the diagram should be implemented:

  - add_item method (in all classes that have that method).
- Add unit tests for all implemented methods.

Reply:
Thanks @jmonto55 , I am working on it😉.